pkg/auth/metadata.go

```diff
@@ -206,13 +212,43 @@ func (m *KVMetadataManager) IsInitialized(ctx context.Context) (bool, error) {
 	return !setupTimestamp.IsZero(), nil
 }
 
+// DockerFileExists For testing purposes
+var DockerFileExists = "/.dockerenv"
+
+func getInstrumentation(metadata map[string]string) {
+	_, k8s := os.LookupEnv("KUBERNETES_SERVICE_HOST")
+	metadata["is_k8s"] = strconv.FormatBool(k8s)
+
+	_, statErr := func() (os.FileInfo, error) {
+		if DockerFileExists == "" {
+			return nil, nil
+		}
+		return os.Stat(DockerFileExists)
+	}()
+	metadata["is_docker"] = strconv.FormatBool(statErr == nil)
+
+	lakefsAccessKeyID := os.Getenv("LAKEFS_ACCESS_KEY_ID")
+	quickstart, _ := strconv.ParseBool(os.Getenv("QUICKSTART"))
+
+	switch {
+	case strings.HasSuffix(lakefsAccessKeyID, "LKFSSAMPLES"):
+		metadata["instrumentation"] = InstrumentationSamplesRepo
+	case quickstart:
+		metadata["instrumentation"] = InstrumentationQuickstart
+	default:
+		metadata["instrumentation"] = InstrumentationRun
+	}
+}
+
 func (m *KVMetadataManager) GetMetadata(ctx context.Context) (map[string]string, error) {
 	metadata := make(map[string]string)
 	metadata["lakefs_version"] = m.version
 	metadata["lakefs_kv_type"] = m.kvType
 	metadata["golang_version"] = runtime.Version()
 	metadata["architecture"] = runtime.GOARCH
 	metadata["os"] = runtime.GOOS
+	getInstrumentation(metadata)
```

Comment:
Suggest to split into functions

```suggestion
	metadata["is_k8s"] = inK8sMetadata()
	metadata["is_docker"] = inDockerMetadata()
	metadata["instrumentation"] = instrumentationMetadata()
```

```go
func isK8sMetadata() string {
	_, k8s := os.LookupEnv("KUBERNETES_SERVICE_HOST")
	return strconv.FormatBool(k8s)
}

func isDockerMetadata() string {
	var err error
	if DockerFileExists != "" {
		_, err = os.Stat(DockerFileExists)
	}
	return strconv.FormatBool(err == nil)
}

func instrumentationMetadata() string {
	lakefsAccessKeyID := os.Getenv("LAKEFS_ACCESS_KEY_ID")
	if strings.HasSuffix(lakefsAccessKeyID, "LKFSSAMPLES") {
		return InstrumentationSamplesRepo
	}
	quickstart, _ := strconv.ParseBool(os.Getenv("QUICKSTART"))
	if quickstart {
		return InstrumentationQuickstart
	}
	return InstrumentationRun
}
```